components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -154,14 +163,56 @@ private void logOperationExecutionResults(ActionType actionType,
     }
 
     @Override
-    public ActionExecutionStatus processErrorResponse(Map<String, Object> map, Event event,
+    public ActionExecutionStatus<Failure> processFailureResponse(Map<String, Object> eventContext,
+                                                        Event actionEvent,
+                                                        ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        processServerError(failureResponse.getFailureReason());
+        return new FailedStatus(new Failure(failureResponse.getFailureReason(),
+                failureResponse.getFailureDescription()));
+    }
+
+    private void processServerError(String errorCode)
+            throws ActionExecutionResponseProcessorException {
+
+        if (isServerError(errorCode)) {
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                        ActionExecutionLogConstants.ACTION_EXECUTION_COMPONENT_ID,
+                        ActionExecutionLogConstants.ActionIDs.EXECUTE_ACTION_OPERATIONS);
+                diagLogBuilder
+                        .resultMessage("An exception was thrown for action. FAILED status should not be used to " +
+                                "process server errors.")
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                        .build();
+                LoggerUtils.triggerDiagnosticLogEvent(diagLogBuilder);
+            }
+            throw new ActionExecutionResponseProcessorException("FAILED status should not be used to process" +
+                    " server errors.");
+        }
```

Comment:
Not clear what you are trying to do here

Reply:
As we discussed, since we are processing the internal server error under the ERROR status, sending the above status with FAILED status is a violation of the contract; therefore, an exception is thrown.